src/sumcheck/bind.rs

```diff
@@ -26,6 +26,19 @@ pub trait SumcheckArray<FieldElement>: Sized {
     // TODO: provide in-place version?
     fn bind(&self, binding: &[FieldElement]) -> Self;
 
+    /// Same as [`Self::bind`], but asserts that this array has not already been reduced to a single
+    /// element.
+    fn bind_assert(&self, binding: &[FieldElement]) -> Self {
+        // Specification interpretation verification: we expect to be binding down to a single
+        // element, but no further.
+        assert!(
+            self.outer_dimension_len() > 1,
+            "binding over a dimension that's already reduced to a single element"
+        );
```

Comment:
If we first guard this assert with `!binding.is_empty()`, we could always use `bind_assert()` in the sumcheck module.